MetadataExtractor/IO/BufferReader.cs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Drew Noakes and contributors. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System.Buffers.Binary;
+
+namespace MetadataExtractor.IO;
+
+internal ref struct BufferReader(ReadOnlySpan<byte> bytes, bool isBigEndian)
+{
+    private readonly ReadOnlySpan<byte> _bytes = bytes;
+    private int _position = 0;
+    private bool _isBigEndian = isBigEndian;
+
+    public long Position => _position;
```

Comment:
Consider returning int here. We don't really support long values.

Reply:
Agreed. This was modeled after SequentialByteReader, but this only works on 32 bit spans.

Reply:
Arrays don't support long indexing, but Streams do. Some image files can be over 2GB, so it's important that we handle long streams correctly.